apps/chat/src/components/Folder/Folder.tsx

```diff
@@ -229,11 +238,41 @@ const Folder = <T extends ConversationInfo | PromptInfo | DialFile>({
     if (!onRenameFolder) {
       return;
     }
-    renameValue.trim() && onRenameFolder(renameValue, currentFolder.id);
+
+    const preparedName = prepareEntityName(renameValue);
+    setRenameValue(preparedName);
+
+    if (
+      !isEntityNameOnSameLevelUnique(
+        preparedName,
+        currentFolder,
+        allFoldersWithoutFilters,
+      )
+    ) {
+      dispatch(
+        UIActions.showToast({
+          message: t(
+            `Folder with name "${preparedName}" already exists in this folder.`,
```

Comment:
Please check how it is implemented in file manager component - your translation will not work properly with current approach